tests/index.ts

```diff
@@ -1,723 +1,555 @@
-		const oldHistory = Promise.resolve(ImmutableList<Block>([new MockBlock(0x7777)]));
-		const newBlock = new MockBlock(0x7777);
-
-		const newHistory = await reconcileBlockHistory(getBlockByHashFactory(), oldHistory, newBlock, onBlockAdded, onBlockRemoved);
-
-		expect(newHistory).to.equal(await oldHistory);
-		expect(newBlockAnnouncements).to.be.empty;
-		expect(blockRemovalAnnouncments).to.be.empty;
-	});
-
-	it("announces a new head when nth block is added to history", async () => {
-		const oldHistory = Promise.resolve(ImmutableList<Block>([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778)
-		]));
-		const newBlock = new MockBlock(0x7779);
-
-		const newHistory = await reconcileBlockHistory(getBlockByHashFactory(), oldHistory, newBlock, onBlockAdded, onBlockRemoved);
-
-		expect(newHistory.toJS()).to.deep.equal([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778),
-			new MockBlock(0x7779)
-		]);
-		expect(newBlockAnnouncements).to.deep.equal([newBlock]);
-		expect(blockRemovalAnnouncments).to.be.empty;
-	});
-
-	it("ignores blocks already in history", async () => {
-		const oldHistory = Promise.resolve(ImmutableList<Block>([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778),
-			new MockBlock(0x7779)
-		]));
-		const newBlock = new MockBlock(0x7778);
-
-		const newHistory = await reconcileBlockHistory(getBlockByHashFactory(), oldHistory, newBlock, onBlockAdded, onBlockRemoved);
-
-		expect(newHistory).to.equal(await oldHistory);
-		expect(newBlockAnnouncements).to.be.empty;
-		expect(blockRemovalAnnouncments).to.be.empty;
-	});
-
-	it("does a multi-block rollback to attach new block to head", async () => {
-		const oldHistory = Promise.resolve(ImmutableList<Block>([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778),
-			new MockBlock(0x7779),
-			new MockBlock(0x777A)
-		]));
-		const newBlock = new MockBlock(0x7779, "BBBB", "AAAA");
-
-		const newHistory = await reconcileBlockHistory(getBlockByHashFactory(), oldHistory, newBlock, onBlockAdded, onBlockRemoved);
-
-		expect(newHistory.toJS()).to.deep.equal([
-			new MockBlock(0x7777, "AAAA"),
-			new MockBlock(0x7778, "AAAA"),
-			new MockBlock(0x7779, "BBBB", "AAAA")
-		]);
-		expect(newHistory.count()).to.equal(3);
-		expect(newBlockAnnouncements).to.deep.equal([newBlock]);
-		expect(blockRemovalAnnouncments).to.deep.equal([
-			new MockBlock(0x777A),
-			new MockBlock(0x7779)
-		]);
-	});
-
-	it("backfills missing blocks", async () => {
-		const oldHistory = Promise.resolve(ImmutableList<Block>([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778)
-		]));
-		const newBlock = new MockBlock(0x777B);
-
-		const newHistory = await reconcileBlockHistory(getBlockByHashFactory(), oldHistory, newBlock, onBlockAdded, onBlockRemoved);
-
-		expect(newHistory.toJS()).to.deep.equal([
-			new MockBlock(0x7777),
-			new MockBlock(0x7778),
-			new MockBlock(0x7779),
-			new MockBlock(0x777A),
-			new MockBlock(0x777B)
-		]);
```

Comment:
I would _really_ like a way to get these tests formatted the way they were.  They are incredibly hard to read without the formatting as it was, and with the formatting this way they become much easier to read (and troubleshoot).  Having spent a lot of time pouring over these tests trying to fix bugs I can say that the formatting plays a very large role in readability and thus debugability, which is why I landed on the format I did.

Note: This sort of thing is why I'm really hesitant to use an auto-formatter, because there are exceptions that a human sometimes needs to make in order to have the code be readable.  While the rule for this is reasonable elsewhere, it may not be reasonable here.

Reply:
We could add this file to `.npmignore`

Reply:
Added and reverted to original.

Reply:
Have you pushed your latest change?  I'm not seeing this reverted per your recent comment.

Reply:
Ah good catch, just pushed.